lib/elements/mixer.ex

```diff
@@ -0,0 +1,105 @@
+defmodule Basic.Elements.Mixer do
+  @moduledoc """
+  Element responsible for mixing the frames coming from two sources, basing on their timestamps.
+  """
+  use Membrane.Filter
+  alias Basic.Formats.Frame
+
+  def_input_pad :first_input,
+    flow_control: :manual,
+    demand_unit: :buffers,
+    accepted_format: %Frame{encoding: :utf8}
+
+  def_input_pad :second_input,
+    flow_control: :manual,
+    demand_unit: :buffers,
+    accepted_format: %Frame{encoding: :utf8}
+
+  def_output_pad :output,
+    flow_control: :manual,
+    accepted_format: %Frame{encoding: :utf8}
+
+  defmodule Track do
+    @type t :: %__MODULE__{
+            buffer: Membrane.Buffer.t(),
+            status: :started | :finished
+          }
+    defstruct buffer: nil, status: :started
+  end
+
+  @impl true
+  def handle_init(_context, _options) do
+    {[],
+     %{
+       tracks: %{first_input: %Track{}, second_input: %Track{}}
+     }}
+  end
+
+  @impl true
+  def handle_buffer(pad, buffer, _context, state) do
+    tracks = Map.update!(state.tracks, pad, &%Track{&1 | buffer: buffer})
+    {tracks, buffer_actions} = get_output_buffers_actions(tracks)
+    state = %{state | tracks: tracks}
+
+    {buffer_actions ++ [redemand: :output], state}
+  end
+
+  @impl true
+  def handle_end_of_stream(pad, _context, state) do
+    tracks = Map.update!(state.tracks, pad, &%Track{&1 | status: :finished})
+    {tracks, buffer_actions} = get_output_buffers_actions(tracks)
+    state = %{state | tracks: tracks}
+
+    if Enum.all?(tracks, fn {_track_id, track} ->
+         track.status == :finished and track.buffer == nil
+       end) do
+      {buffer_actions ++ [end_of_stream: :output], state}
+    else
+      {buffer_actions ++ [redemand: :output], state}
+    end
+  end
+
+  @impl true
+  def handle_demand(:output, _size, _unit, context, state) do
+    demand_actions =
+      state.tracks
+      |> Enum.filter(fn {track_id, track} ->
+        track.status != :finished and track.buffer == nil and context.pads[track_id].demand == 0
+      end)
+      |> Enum.map(fn {track_id, _track} -> {:demand, {track_id, 1}} end)
+
+    {demand_actions, state}
+  end
+
+  defp has_buffer?({_track_id, track}),
+    do: track.buffer != nil
+
+  defp can_send_buffer?(tracks) do
+    active_tracks =
+      tracks
+      |> Enum.filter(fn {_track_id, track} -> track.status == :started end)
+
+    Enum.all?(active_tracks, &has_buffer?/1) and Enum.any?(tracks, &has_buffer?/1)
```

Comment:
I would opt for rewriting it somehow to emphasize that there are indeed two cases - either `active_tracks` can be empty (and then it's sufficient that any of `tracks` to `has_buffer`) or `active_tracks` might be non empty and in such a case, all `active_tracks` must have a buffer).
How about something like: 
```
(active_tracks == [] and Enum.any?(tracks, &has_buffer?/1)) or (active_tracks != [] and Enum.all?(active_tracks, &has_buffer?/1))
```